src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/SecretUtils.java

```diff
@@ -97,6 +101,18 @@ public static String base64DecodeToString(String s) {
         return null;
     }
 
+    /**
+     * Get the scope from a given {@code Secret}.
+     * @param s the secret whose scope we want to obtain.
+     * @return the scope for a given secret.
+     */
+    public static CredentialsScope getCredentialScope(Secret s) {
+        return CredentialsScope.valueOf(Optional
```

Comment:
should there be a check the scope is valid before attempting to use it, to give some better logging?
if someone has a typo on 1 of their thousand secrets would they be able to diagnose it correctly?

Reply:
Yeah, you're right. I think a warning log message should be enough don't you think?

Reply:
Log should be fine containing the secret id